### PR TITLE
workspaces: add validation function for the workflows 

### PR DESCRIPTION
**Pull request by marcdiazsan**
Goes on top of reanahub/reana-client#545 PR set, and reanahub/reana#532 PR set

Closes reanahub/reana-client#546

### CODE REVIEW

File: reana_commons/config.py [hunk 1]
```diff
@@ -336,6 +336,11 @@ def kubernetes_node_label_to_dict(node_label):
 SHARED_VOLUME_PATH = os.getenv("SHARED_VOLUME_PATH", "/var/reana")
 """Default shared volume path."""
 
+DEFAULT_WORKSPACE_PATH = os.getenv("DEFAULT_WORKSPACE_PATH", "/var/reana")
+"""Default workspace path defined by the admin."""
+
+ADMIN_ALLOWED_WORKSPACES = os.getenv("ADMIN_ALLOWED_WORKSPACES", None)
```
Comment (audrium) [inline reana_commons/config.py:342]: Minor: docstring is missing.

Perhaps we could split the string and append `SHARED_VOLUME_PATH` here to have the final list. In that case code will be simplified `reana_commons/workspaces.py` and this operation will be done only once

Comment (audrium) [inline reana_commons/config.py:342]: Also, please see [my comment](https://github.com/reanahub/reana/pull/533/files#r700897170) about the name

**Review by audrium — COMMENTED** [review]
Few minor comments, it all works fine!

File: reana_commons/config.py (hunk 1 above)
Comment (marcdiazsan) [reply reana_commons/config.py:342]: I agree, good change, as I wont have to make that operation in the validation or showing the workspaces

File: reana_commons/config.py [hunk 2]
```diff
@@ -336,6 +336,23 @@ def kubernetes_node_label_to_dict(node_label):
...
+"""Default workspace path defined by the admin."""
+
+
+def workspaces(workspaces_list):
+    """Load the available workspaces as list."""
+    if workspaces_list:
+        workspaces_list = workspaces_list.split(",")
+        workspaces_list.append(SHARED_VOLUME_PATH)
+    else:
+        workspaces_list = [SHARED_VOLUME_PATH]
```
Comment (audrium) [inline reana_commons/config.py:349]: Very minor: could be simplified like this
```python
workspaces_list = workspaces_list.split(",") if workspaces_list else []
workspaces_list.append(SHARED_VOLUME_PATH)
```

File: reana_commons/workspaces.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+
+def validate_workspace(workspace_option):
+    """Validate and return workspace.
+
+    :param workspace_option: A string of the workspace.
+    :returns: A string of the validated workspace.
+    """
+    if workspace_option:
+        if workspace_option not in WORKSPACE_PATHS:
```
Comment (audrium) [inline reana_commons/workspaces.py:21]: We need to take into account that one of the `WORKSPACE_PATHS` could be for example `/eos`,
but the user defined workspace would be `/eos/user/a/amecioni/workflows/my_super_workflow.` So direct comparison would not allow this workspace definition even though it's perfectly valid. I guess for now we could iterate through the `WORKSPACE_PATHS` list and check if `workspace_option` `startswith()` at least one path?

Later some more complex validation could be done (better in a separate PR in the future) to catch all the possible ways to abuse the system, like to catch "../", etc. (thanks @mvidalgarcia for noticing this possibility)

**Review by audrium — APPROVED** [review]
Nicely done!

File: reana_commons/workspaces.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+def validate_workspace(workspace_option, available_paths=WORKSPACE_PATHS):
+    """Validate and return workspace.
+
+    :param workspace_option: A string of the workspace.
+    :returns: A string of the validated workspace.
+    """
+    if workspace_option:
+        available = any(
+            workspace_option.startswith(os.path.abspath(path))
```
Comment (audrium) [inline reana_commons/workspaces.py:23]: Minor: I guess `workspace_option` could be also wrapped with `os.path.abspath()` to catch cases where users are trying to access non allowed locations.
Example: `workspace_option="/eos/../otherdir"`

**Pull request merged**